engines/mohawk/riven.cpp

```diff
@@ -761,23 +756,10 @@ void MohawkEngine_Riven::saveGameStateAndDisplayError(int slot, const Common::St
 	}
 }
 
-void MohawkEngine_Riven::tryAutoSaving() {
-	if (!canSaveGameStateCurrently() || _gameEnded) {
-		return; // Can't save right now, try again on the next frame
-	}
-
-	_lastSaveTime = _system->getMillis();
-
-	if (!_saveLoad->isAutoSaveAllowed()) {
-		return; // Can't autosave ever, try again after the next autosave delay
-	}
-
-	Common::Error saveError = saveGameState(RivenSaveLoad::kAutoSaveSlot, "Autosave", true);
-	if (saveError.getCode() != Common::kNoError)
-		warning("Attempt to autosave has failed.");
+bool MohawkEngine_Riven::canSaveAutosaveCurrently() {
+	return canSaveGameStateCurrently() && !_gameEnded;
```

Comment:
`_saveLoad->isAutoSaveAllowed()` is no longer called?

The thing is in Myst and Riven, we check whether the save in the autosave slot can be overwritten by the autosave (maybe the user somehow put their own save in there). To do so, we open the save and check if it has the autosave flag set.
Reading the file is a somewhat costly operation, and we were careful to do it at most once per autosave interval. Now, in the worst case, it is done every frame and for each event in the queue. So maybe `canSaveAutosaveCurrently` should allow to tell the autosave system "no, autosaving is not possible currently, and don't try again immediately".

Reply:
Actually, that's not the case. The way I set things up in Engine is that it waits for the autosave interval to expire, and only then calls saveAutosaveIfEnabled. It's there that I check canAutosaveCurrently, and even if it returns false, I still reset the autosave interval.

But in retrospect, it does have the downside that if autosaves were, say, set for half a hour, if you just happen to have a dialog or cutscene open on the exact minute the autosave check triggers, it'd wait another half hour before it tried saving again. What I think I'll do is have that if it can't do the autosave, it'll reset the next interval to try in 5 minutes time, irrespective of what the normal autosave interval is.

Reply:
Oh, right, I mixed things up with the previous behavior, sorry.

That call to `_saveLoad->isAutoSaveAllowed()` still needs to be restore though.

Reply:
Sorry I forgot it. Fixed.